lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -2465,33 +2766,34 @@ struct ServerDriveQueryDirectoryRequest {
 
 impl ServerDriveQueryDirectoryRequest {
     fn decode(device_io_request: DeviceIoRequest, payload: &mut Payload) -> RdpResult<Self> {
-        let fs_information_class_lvl =
-            FsInformationClassLevel::from_u32(payload.read_u32::<LittleEndian>()?)
-                .ok_or_else(|| invalid_data_error("failed to read FsInformationClassLevel"))?;
-        if fs_information_class_lvl != FsInformationClassLevel::FileDirectoryInformation
-            && fs_information_class_lvl != FsInformationClassLevel::FileFullDirectoryInformation
-            && fs_information_class_lvl != FsInformationClassLevel::FileBothDirectoryInformation
-            && fs_information_class_lvl != FsInformationClassLevel::FileNamesInformation
-        {
+        let file_info_class_lvl =
+            FileInformationClassLevel::from_u32(payload.read_u32::<LittleEndian>()?)
+                .ok_or_else(|| invalid_data_error("failed to read FileInformationClassLevel"))?;
+
+        // These are all the FsInformationClass's supported for this message by FreeRDP
+        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L794
+        let valid_levels = vec![
+            FileInformationClassLevel::FileDirectoryInformation,
+            FileInformationClassLevel::FileFullDirectoryInformation,
+            FileInformationClassLevel::FileBothDirectoryInformation,
+            FileInformationClassLevel::FileNamesInformation,
+        ];
+
+        if !valid_levels.contains(&file_info_class_lvl) {
             return Err(invalid_data_error(&format!(
-                "read invalid FsInformationClassLevel: {:?}, expected one of {:?}",
-                fs_information_class_lvl,
-                vec![
-                    FsInformationClassLevel::FileDirectoryInformation,
-                    FsInformationClassLevel::FileFullDirectoryInformation,
-                    FsInformationClassLevel::FileBothDirectoryInformation,
-                    FsInformationClassLevel::FileNamesInformation
-                ]
+                "read invalid FileInformationClassLevel: {:?}, expected one of {:?}",
+                file_info_class_lvl, valid_levels,
             )));
         }
+
         let initial_query = payload.read_u8()?;
         let mut path_length: u32 = 0;
         let mut path = String::from("");
+        let mut padding: [u8; 23] = [0; 23];
         if initial_query != 0 {
             path_length = payload.read_u32::<LittleEndian>()?;
 
             // TODO(isaiah): make a payload.skip(n)
-            let mut padding: [u8; 23] = [0; 23];
             payload.read_exact(&mut padding)?;
```

Comment:
```suggestion
            payload.seek(SeekFrom::Current(23))?;
```
Make sure you import the [Seek](https://doc.rust-lang.org/stable/std/io/trait.Seek.html) trait from the `std::io` as it allow you to use the `seek` method. 

By doing it this way you can avoid using the padding field in the struct and also remove it from here.

Reply:
Added as an issue to go back and do once our stack of PR's is all merged.

https://github.com/gravitational/teleport/issues/13990